docs/styles.rst

```diff
@@ -0,0 +1,128 @@
+Styles
+======
+
+VSG supports several predefined styles.
+They can be used with the **--style** command line option.
+
+The table below lists the built in styles available
+
++---------------+--------------------------------------------+
+| Style         | Description                                |
++---------------+--------------------------------------------+
+| legacy        | Rules based on the pre 2.0.0 release       |
+| lowercase     | All keywords and identifiers are lowercase |
++---------------+--------------------------------------------+
+
+Style Descriptions
+------------------
+
+legacy
+~~~~~~
+
+This style attempts to improve readability by:
+
+* Emphasising on non vhdl identifiers by capitalizing them.
+
+  * entity names
+  * architecture names
+  * ports
+  * generics
+  * etc...
+
+* Blank lines added between major items
+
+  * processes
+  * if statements
+  * case statements
+
+* Alignments
+
+  * :'s over entire entities, components, instantiations, etc...
+  * <='s over groups of sequential statements
+  * inline comments within processes, architecture declarative regions, etc...
+
+* Indenting
+
+  * 2 spaces
+
+* Structure
+
+  * No single line sequential statements using the when keyword
+  * No code after the case when statements
+  * Split if/elsif/else/end if into separate lines
+  * Removing comments from instantiation and component ports and generics
+  * No more than two signals can be declared on a single line
+
+lowercase
```

Comment:
Do you use this lowercase style? If no, then probably there is no point in adding styles just to have more styles.

Reply:
that was more of an example of what could be done.  

At one point we switched VSG from solely my style to one that is more...hmm...I can't find the word...less my style?.  So the new style became the default.  It would seem we should name it something and then I can name my style something besides legacy.  You could even add your own style.  The way you want to format signal declarations is really interesting and I never would have thought of it.

maybe this feeds back into the other discussion on whether to make `--style` a positional argument?